src/liboslexec/llvm_gen.cpp

```diff
@@ -3583,6 +3583,19 @@ LLVMGEN (llvm_gen_luminance)
 
 
 
+LLVMGEN (llvm_gen_isconstant)
+{
+    Opcode &op (rop.inst()->ops()[opnum]);
+    ASSERT (op.nargs() == 2);
+    Symbol &Result (*rop.opargsym (op, 0));
+    ASSERT (Result.typespec().is_int());
+    Symbol &A (*rop.opargsym (op, 1));
+    rop.llvm_store_value (rop.ll.constant(A.is_constant() ? 1 : 0), Result);
```

Comment:
I should clarify, isconstant(x) is trying to tell you whether the expression is a *known* (to the optimizer) constant value. It is NOT telling you if the expression is *uniform* (having the same value at all points being shaded), which is not the same as knowing its actual value with certainty.